light-zk.js/src/wallet/user.ts

```diff
@@ -396,6 +396,23 @@ export class User {
       : BN_0;
     publicAmountSpl = convertedPublicAmounts.publicAmountSpl;
 
+    if (publicAmountSol && this.provider.connection) {
+      const solBalance = await this.provider.connection.getBalance(
+        this.provider.wallet.publicKey,
+      );
+      if (solBalance < publicAmountSol.toNumber()) {
```

Comment:
`solBalance` has type `Balance` and `publicAmountSol` at this point should be BN. `Balance` has a `totalSolBalance` field of type `BN`. So I think you could rather do:

```suggestion
      if (solBalance.totalSolBalance.lt(publicAmountSol)) {
```

Reply:
solBalance in my case has type `number`, I'm getting it via [solana-web3.js getBalance](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getBalance.getBalance-1).

Reply:
@vadorovsky solBalance is the balance of the solana address not our user shielded balance :)

Reply:
Alright, sorry then

Reply:
> The error is good but the sol is still missing :D
> 
> ```
>      Error: INSUFFICIENT_BAlANCE: The current balance is insufficient for this operation. The user's balance is 18.9302266 SOL, but the operation requires 20 SOL.
> ```

It's for testing purposes :)

Reply:
I think it makes sense to throw meaningful errors early, thanks for adding.
@sergeytimoshin can you increase the amount of airdropped sol or reduce the amount of sol shielded so that tests pass?